src/openrct2/core/FileIndex.hpp

```diff
@@ -183,15 +183,20 @@ class FileIndex
         Console::WriteLine("Building %s (%zu items)", _name.c_str(), scanResult.Files.size());
 
         auto startTime = std::chrono::high_resolution_clock::now();
+        // Start at 1, so that we can reach 100% completion status
+        size_t i = 1;
         for (auto filePath : scanResult.Files)
         {
+            Console::WriteFormat("File %4d of %d, done %3d%%\r", i, scanResult.Files.size(), i * 100 / scanResult.Files.size());
```

Comment:
Why the `\r`?

Reply:
So that the carriage returns and you have a single line with status (continually overwritten) instead of multiple thousands

Reply:
This won't be interpreted as a newline on any of our support OSs, then?

Reply:
No, `\r` works cross-platform, fortunately. This is a fairly common technique.

Reply:
The best way to know would be to check. Android and its logcat would probably output a line for each of the objects.

Reply:
@AaronVanGeffen can you confirm it works the intended way on macOS?

Reply:
macOS has used `\n` for newlines since its tenth version, so `\r` should behave just like it would on Linux.

Reply:
I'd make it `5%d`, some players are likely to have over 10,000 things.